test/unit/files/FilesRepository.test.ts

```diff
@@ -0,0 +1,479 @@
+        const testOffset = 20;
+        const testFileOrderCriteria = FileOrderCriteria.NEWEST;
+
+        const actual = await sut.getDatasetFiles(
+          TestConstants.TEST_DUMMY_PERSISTENT_ID,
+          testVersionId,
+          testLimit,
+          testOffset,
+          testFileOrderCriteria,
+        );
+
+        const expectedRequestConfig = {
+          params: {
+            limit: testLimit,
+            offset: testOffset,
+            orderCriteria: testFileOrderCriteria.toString(),
+          },
+          headers: TestConstants.TEST_EXPECTED_AUTHENTICATED_REQUEST_CONFIG_API_KEY.headers,
+        };
+
+        assert.calledWithExactly(
+          axiosGetStub,
+          `${TestConstants.TEST_API_URL}/datasets/:persistentId/versions/${testVersionId}/files?persistentId=${TestConstants.TEST_DUMMY_PERSISTENT_ID}`,
+          expectedRequestConfig,
+        );
+        assert.match(actual, [testFile]);
+      });
+
+      test('should return error result on error response', async () => {
+        const axiosGetStub = sandbox.stub(axios, 'get').rejects(TestConstants.TEST_ERROR_RESPONSE);
+
+        let error: ReadError = undefined;
+        await sut.getDatasetFiles(TestConstants.TEST_DUMMY_PERSISTENT_ID).catch((e) => (error = e));
+
+        assert.calledWithExactly(
+          axiosGetStub,
+          `${TestConstants.TEST_API_URL}/datasets/:persistentId/versions/:latest/files?persistentId=${TestConstants.TEST_DUMMY_PERSISTENT_ID}`,
+          TestConstants.TEST_EXPECTED_AUTHENTICATED_REQUEST_CONFIG_API_KEY,
+        );
+        expect(error).to.be.instanceOf(Error);
+      });
+    });
+  });
+
+  describe('getFileDownloadCount', () => {
+    const testCount = 1;
+    const testFileDownloadCountResponse = {
+      data: {
+        status: 'OK',
+        data: {
+          message: `${testCount}`,
+        },
+      },
+    };
+
+    describe('by numeric id', () => {
+      test('should return count when providing id and response is successful', async () => {
+        const axiosGetStub = sandbox.stub(axios, 'get').resolves(testFileDownloadCountResponse);
+        const expectedApiEndpoint = `${TestConstants.TEST_API_URL}/files/${testFile.id}/downloadCount`;
+
+        // API Key auth
+        let actual = await sut.getFileDownloadCount(testFile.id);
+
+        assert.calledWithExactly(
+          axiosGetStub,
+          expectedApiEndpoint,
+          TestConstants.TEST_EXPECTED_AUTHENTICATED_REQUEST_CONFIG_API_KEY,
+        );
+        assert.match(actual, testCount);
+
+        // Session cookie auth
+        ApiConfig.init(TestConstants.TEST_API_URL, DataverseApiAuthMechanism.SESSION_COOKIE);
+
+        actual = await sut.getFileDownloadCount(testFile.id);
+
+        assert.calledWithExactly(
+          axiosGetStub,
+          expectedApiEndpoint,
+          TestConstants.TEST_EXPECTED_AUTHENTICATED_REQUEST_CONFIG_SESSION_COOKIE,
+        );
+        assert.match(actual, testCount);
+      });
```

Comment:
seems like a repeating pattern

// API Key auth
test

// Session cookie auth
test

I think is worth developing a way of automatically running any test twice with the 2 configuration, maybe in another issue